internal/core/command/main.go

```diff
@@ -16,43 +16,40 @@ package command
 
 import (
 	"context"
-	"flag"
+	"os"
 
 	"github.com/edgexfoundry/edgex-go"
 	"github.com/edgexfoundry/edgex-go/internal"
 	"github.com/edgexfoundry/edgex-go/internal/core/command/config"
-	container "github.com/edgexfoundry/edgex-go/internal/core/command/container"
+	"github.com/edgexfoundry/edgex-go/internal/core/command/container"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/bootstrap/handlers/database"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/telemetry"
-	"github.com/edgexfoundry/edgex-go/internal/pkg/usage"
 
 	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/commandline"
 	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/handlers/httpserver"
 	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/handlers/message"
 	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/handlers/secret"
 	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/handlers/testing"
 	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/interfaces"
 	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/startup"
 	"github.com/edgexfoundry/go-mod-bootstrap/di"
-
 	"github.com/edgexfoundry/go-mod-core-contracts/clients"
-
 	"github.com/gorilla/mux"
 )
 
 func Main(ctx context.Context, cancel context.CancelFunc, router *mux.Router, readyStream chan<- bool) {
 	startupTimer := startup.NewStartUpTimer(internal.BootRetrySecondsDefault, internal.BootTimeoutSecondsDefault)
 
-	var useRegistry bool
-	var configDir, profileDir string
-
-	flag.BoolVar(&useRegistry, "registry", false, "Indicates the service should use registry.")
-	flag.BoolVar(&useRegistry, "r", false, "Indicates the service should use registry.")
-	flag.StringVar(&profileDir, "profile", "", "Specify a profile other than default.")
-	flag.StringVar(&profileDir, "p", "", "Specify a profile other than default.")
-	flag.StringVar(&configDir, "confdir", "", "Specify local configuration directory")
-	flag.Usage = usage.HelpCallback
-	flag.Parse()
+	// All common command-line flags have been moved to DefaultCommonFlags. Service specific flags can be add here,
+	// by inserting service specific flag prior to call to commonFlags.Parse().
+	// Example:
+	// 		flags.FlagSet.StringVar(&myvar, "m", "", "Specify a ....")
+	//      ....
+	//      flags.Parse(os.Args[1:])
+	//
+	flags := commandline.NewDefaultCommonFlags("")
```

Comment:
Nitpick - prefer alternate factory function for use-case of "no usage"; i.e. ```commandLine.NewDefaultCommonFlagsWithoutUsage()```.  More explicitly expresses intent than passing in an empty string to the existing factory function.  Don't have to change; think doing so will make code tiny bit better.

Reply:
I see your point, but this is in go-mod-bootstrap which would require another round of PRs.